### ISSUE TITLE
Checkbox: state should be read out after change

### ISSUE THREAD

**Issue by AccessibilityTestingTeam-TCS** [Microsoft/vscode#52274]
**Environment Details:**
VSCode Version : 1.24.0
OS Version : Win10

**Additional Details:** 
MAS Violated : MAS 3.2.2 , MAS 1.3.1 , MAS 2.4.3
ScreenReader Versions : NVDA2018.1.1 / JAWS18.0.5038

**Repro Steps:**
1. Launch VS Code while NVDA/JAWS is ON.
2. Navigate to Activity Bar and select "Search"(Cntrl+Shift+F) button.
3. Navigate to "Match Case" , "Match Whole Word" , "Use RegEx","Use Exclude settings and Ignore Files", "Replace All" comboboxes  using "Tab".
4. Change the state of the these controls using "Space Bar".

**Actual:**
NVDA does not read the state of the comboboxes as soon as any action is performed on them.

Focus moves to start of the edit field - "Search"/"Files to exclude" for corresponding checkboxes. Moving focus disrupts the tab order.

Without any user's consent the focus moves to start of edit field and screenreader does not read the state changed by the user for any combobox.

**Same issue has been observed with JAWS.**

**Expected:**
NVDA should read the state of the combobox as soon as any action is performed by the user on it and also focus should not move to edit field when any action is performed. It should remain on the combobox on which the action was performed. 

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx

**User Impact:**
Screenreader users will not be able to know whether the performed action(unchecking/unchecking) was successful or not.

Keyboard only users will need to navigate again and again to checkboxes on which they have already performed action so as to navigate to another checkbox/control in sequence.

**MAS Reference:**
MAS 3.2.2 - https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={24cbfff7-374e-42e6-b9ab-d87fcd53b3e8}

MAS 1.3.1 - https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

MAS 2.4.3 - https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0de7fbe1-ad7e-48e5-bcbb-8d986691e2b9}

**Attachment For Reference:**
[A11y_VSCode_ViewSearch_ScreenReader_CheckboxState.pptx](https://github.com/Microsoft/vscode/files/2114234/A11y_VSCode_ViewSearch_ScreenReader_CheckboxState.pptx)

**NOTE :** This issue occurs intermittently. Need to try several times to repro this issue.

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
@isidorn this is because when I use NVDA and press 'space' on a button, the browser actually gets a click event, and we only fix the focus when we think the use clicked.

I remember debugging this before and I thought there was some discussion on another accessibility issue, but I've searched all over and can't find it. Do you know whether someone has looked at this already?

My proposal is to just not move focus here when screenreader mode is enabled.

**Comment by roblourens** [comment]
Ping @isidorn is it ok to change the behavior based on screenreader mode like that?

**Comment by roblourens** [comment]
Oh the other issue I was thinking of was https://github.com/Microsoft/vscode/issues/52590 (it was missing the accessibility tag earlier)

**Comment by isidorn** [comment]
@roblourens sorry for the slow response I was on vacation.
As I mentioned in the other issue we try to not to do things differerently based on if a screen reader is turned on or not. However since the change is minor we could introduce that as a workaround - if we have no better way to solve this.

**Comment by isidorn** [comment]
Assigning to september since we need this for the better accessibility grade. So we try to find some solution, feel free to move out to October

**Comment by roblourens** [comment]
I'll try that, FYI @rebornix this also affects the find widget, don't know if you have a better idea.

**Comment by chrmarti** [comment]
It does not move the focus when not in screen reader mode and using space to toggle for me. Whether or not screen reader mode is on only affects mouse clicks in my setup. In that sense I can't reproduce the problem, please add instructions.

**Comment by roblourens** [comment]
Verify:

Screenreader on:
- Clicking or pressing spacebar on the toggle button keeps focus on the toggle button and reads the status in some way

Screenreader off:
- Clicking moves focus to the input button
- Toggling with space does not move focus
